### PR TITLE
add task 8 implementation

### PR DESCRIPTION
**Pull request by AlbertHaikou**
Task 8.1 is implemented

### CODE REVIEW

**Comment by maksumov** [comment]
> Task 8.1 is implemented

Sorry, but I can't test anything except building the docker image and Dockerfile is non-working.

How I tested:
```bash
git clone git@github.com:AlbertHaikou/nodejs-aws-be.git
cd nodejs-aws-be/
git fetch origin task8:task8
git checkout task8
cd cart-service/
npm install
docker build -t cart-api -f dockerfiles/Dockerfile .
```

docker output:
```
Sending build context to Docker daemon  456.7kB
Step 1/13 : FROM node:12-alpine as build
 ---> 844f8bb6a3f8
Step 2/13 : WORKDIR /app
 ---> Using cache
 ---> f855fb2aff25
Step 3/13 : COPY ./package*.json ./
 ---> 5da933af7cba
Step 4/13 : RUN npm install
 ---> Running in 66b8f63875ef

> @nestjs/core@7.5.5 postinstall /app/node_modules/@nestjs/core
> opencollective || exit 0

                           Thanks for installing nest 
                 Please consider donating to our open collective
                        to help us maintain this package.
                                         
                            Number of contributors: 0
                              Number of backers: 469
                              Annual budget: $42,435
                             Current balance: $4,993
                                         
             Become a partner: https://opencollective.com/nest/donate
                                         
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

added 1036 packages from 661 contributors and audited 1041 packages in 38.012s

72 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

Removing intermediate container 66b8f63875ef
 ---> 86a6d246b892
Step 5/13 : COPY . ./
 ---> c48d1fedf23d
Step 6/13 : RUN npm run build
 ---> Running in f911715969dc

> rs-checkout-api@0.0.1 prebuild /app
> rimraf dist


> rs-checkout-api@0.0.1 build /app
> nest build

Removing intermediate container f911715969dc
 ---> aa59db316acd
Step 7/13 : FROM node:12-alpine
 ---> 844f8bb6a3f8
Step 8/13 : WORKDIR '/app'
 ---> Using cache
 ---> f855fb2aff25
Step 9/13 : COPY --from=build /project/dist /project/dist
COPY failed: stat /var/lib/docker/overlay2/0b68f7c0966b9c043ef882bcdd4a152d718df84c5a1a758f2983df59b79beffe/merged/project/dist: no such file or directory
```

File: cart-service/dockerfiles/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+WORKDIR /app
+COPY ./package*.json ./
+RUN npm install
+COPY . ./
+RUN npm run build
+
+
+FROM node:12-alpine
+WORKDIR '/app'
+COPY --from=build /project/dist /project/dist
```
Comment (maksumov) [inline cart-service/dockerfiles/Dockerfile]: At this line docker break work with message that **/project/dist** folder doesn't exist.

Comment (AlbertHaikou) [reply cart-service/dockerfiles/Dockerfile]: Thanks, yes, sorry, I forgot to commit and push the latest version, now it is fixed.

Comment (maksumov) [reply cart-service/dockerfiles/Dockerfile]: Ok. now it works
image

**Comment by maksumov** [comment]
- [x] Dockerfile is prepared, image is building. Image size is minimised to be less than 500 MB.
image
- [x] Dockerfile is optimized. Files that change more often and commands that depend on them should be included later, files and commands that change less should be at the top.
- [ ] Folders are added to .dockerignore, **with explanations**. **At least 2 big directories should be excluded from build context. Elastic Beanstalk application is initialized.**
- [ ] **Environment is created and the app is deployed to the AWS cloud.** You must provide a link to your GitHub repo with Cart API service or PR with created Dockerfile and related configurations.
- [ ] **FE application is updated with Cart API endpoint. You must provide a PR with updates in your FE repository and OPTIONALLY link to deployed front-end app which makes proper API calls to your Cart service.**

Current score: 2/5

**Pull request merged**